src/schematics/ng-add/index.ts

```diff
@@ -0,0 +1,102 @@
+import { NgAddSchematicsSchema } from './schema';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { chain, Rule, SchematicContext, SchematicsException, Tree } from '@angular-devkit/schematics';
+import { addPackageJsonDependency, getPackageJsonDependency, NodeDependencyType } from '@schematics/angular/utility/dependencies';
+import * as semver from 'semver';
+
+export function install(tree: Tree, context: SchematicContext): Tree {
+  const installOptions = process.env && process.env.npm_execpath && process.env.npm_execpath.indexOf('yarn') === -1 ? {} : { packageManager: 'yarn' };
+  context.addTask(new NodePackageInstallTask(installOptions));
+  return tree;
+}
+
+function readPackageJson(tree: Tree, workspaceProject: any) {
+  // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+  const workspaceConfig = tree.read(`${workspaceProject.root}/package.json`);
+  if (!workspaceConfig) {
+    throw new SchematicsException('Could not find package.json file');
+  }
+  return JSON.parse(workspaceConfig.toString());
+}
+
+function readAngularJson(tree: Tree) {
+  const workspaceConfig = tree.read('/angular.json');
+  if (!workspaceConfig) {
+    throw new SchematicsException('Could not find Angular workspace configuration');
+  }
+  return JSON.parse(workspaceConfig.toString());
+}
+
+function getFinalProjectName(workspace: any, projectName: string) {
+  if (projectName.length > 0) {
+    return projectName;
+  } else if (Object.keys(workspace.projects).length !== 1) {
+    throw new SchematicsException('No project name provided and several projects were found in angular.json.');
+  } else {
+    return Object.keys(workspace.projects)[0];
+  }
+}
+
+/**
+ * Update the application with the prefetch script
+ *
+ * @param options
+ */
+function updatePrefetchBuilder(options: NgAddSchematicsSchema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+
+    const workspace = readAngularJson(tree);
+    const projectName = getFinalProjectName(workspace, options.projectName);
+
+    const workspaceProject = workspace.projects[projectName];
+    const packageJson = readPackageJson(tree, workspaceProject);
+
+    if (!packageJson.scripts) {
```

Comment:
`packageJson.scripts ||= {}`

Reply:
done, as well for `workspaceProject.architect ||= {};` below